### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by gavrochelegnou** [hunzaboy/Light-Modal#1]
Nice work, 

But for the record this kind of work lacks all the accessibilty features provided by common modal scripts.
And without JS it will never be able to have all the mandatory features to pass any accessibilty validation (WCAG, section 508, ...) .

I think this should be emphasized on the "Readme/how to use" to avoid this kind of "proof of concept" project to go live on any website where it could prevent real users from accessing important content.

**Comment by hunzaboy** [comment]
What kind of accessibility features is missing? I would love to integrate them. Any good references will be helpful.

**Comment by thiemeljiri** [comment]
Hey,

First, I'd like to say that this is really nice usage of `:target` selector. The default anchor browser behaviour takes care of the history tracking and ability to open the lightbox when the specific URL is called. It also sets the default focus on the modal.

All this is great start to enhance from here. However the enhancement cannot be done HTML/CSS only.

The user shouldn't be able to set focus to the elments outside of the lightbox. Also, everything except for the lightbox should have also `aria-hidden="true"` set. Keeping the focus inside of the lightbox is probably the most complicated part. I prefer to have the lightboxes outside of the main container of the page. When lightbox is opened I set `aria-hidden="true"` to the main container. As for the focus I watched for keyboard or focusin events and whether any lightbox was opened. But next time I'll probably use `inert` attribute. It's not yet even in a draft afaik but has several polyfils. I think this idea won't be abandoned and once implemented it it will also prevent the browsers from searching inside the inert areas (which I don't think can be done any other way). But feel free to find any solution for the focus management you like. There are multiple.

After closing the lightbox the focus should be set back to the opener button. Currently it gets you back to the beginning of the page. This also scrolls the page to the top after closing the lightbox which is not obvious from the demo. This is not even accessibility but also UX issue.

Which is probably not critical but users are used for that, is the possibility to close the lightbox with Esc key. I myself use this always as the first option because I prefer keyboard when possible. When the devoleper disappoints me, only then I reach for the mouse.

This are the main requirements for the accessibility of the ligthboxes which your solution lacks. But implementing them will mean that you will have to remove "CSS-only" from the name. But accessible CSS-only lightbox doesn't exist. It's not possible. At least in the near future.

Keep up the good work.

Cheers,
Jiri

**Comment by hunzaboy** [comment]
Thank @thiemeljiri for such a detailed explanation. I think I need to accompany it with a js file, to overcome the accessibility issue, as an addon to the CSS version.

**Comment by napengam** [comment]
You need to controll keyboard input, like TAB.
In your demo, with the TAB-key, I am able to access any functionality behind the 'modal' dialog box.
This is probably not what the user should be able to do !
  
Regards
heinz

**Comment by gavrochelegnou** [comment]
Hello, 
Thanks everyone for the great comments.

I think the only thing you could improve in pure HTML/CSS is : 
- Remove the ` aria-hidden='true' ` attribute. It  completelty prevents screenreaders from accessing the content of the modal.
- Replace it with a css visibility:hidden value that you can change to visibility:visible once the modal receives focus.

For the rest as detailed by @thiemeljiri  you will need to add some JS code.

You can find a summary of what to do here : 
https://clearleft.com/posts/416

And some more detailed specifications for dialogs here : 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role
and here :
https://www.marcozehe.de/2015/02/05/advanced-aria-tip-2-accessible-modal-dialogs/

And some notes here :
https://www.w3.org/WAI/GL/wiki/Using_ARIA_role%3Ddialog_to_implement_a_modal_dialog_box

Cheers,

**Comment by hunzaboy** [comment]
@gavrochelegnou , thanks for the info. I will be working on making the modal more accessible based on all the information gathered from the awesome opensource community.

**Comment by hunzaboy** [comment]
Anyone is welcome to submit a PR request for the JS version for full accessibility.
